### PR TITLE
Update ReadMe.md

### PR DESCRIPTION
**Pull request by babaktaremi**
Removed NH 
Added REPR Pattern and minimal APIs in "API" section Added Some Good to know libraries
Added Output Caching and Response Caching docs in "Caching" section

### CODE REVIEW

File: ReadMe.md [hunk 1]
```diff
@@ -117,6 +117,8 @@ If you like or are using this project to learn or start your solution, please gi
        2. [Memcached](https://memcached.org)
     - Entity Framework 2nd Level Cache
        - [EFCoreSecondLevelCacheInterceptor](https://github.com/VahidN/EFCoreSecondLevelCacheInterceptor)
+    - [Response Caching](https://learn.microsoft.com/en-us/aspnet/core/performance/caching/response?view=aspnetcore-7.0)
```
Comment (MoienTajik) [inline ReadMe.md:120]: Can you remove `?view=aspnetcore-7.0` from the URLs? Microsoft redirects these URLs to the latest version by default.

**Review by MoienTajik — CHANGES_REQUESTED** [review]
The roadmap xMind/Image need to be updated as well. I appreciate if you do it, otherwise I can do that.

**Comment by babaktaremi** [comment]
> The roadmap xMind/Image need to be updated as well. I appreciate if you do it, otherwise I can do that.

I'll update it by end of the day and put it in a new PR

**Comment by MoienTajik** [comment]
> I'll update it by end of the day and put it in a new PR

Thanks. You can apply the changes on this PR and will merge all together then.

**Comment by babaktaremi** [comment]
can [Websurge](https://websurge.west-wind.com/) be an option for Performance testing tools?

**Review by kiranshahi — APPROVED** [review]

**Comment by babaktaremi** [comment]
Just Updated xMind/Image
Also Categorized the added things to more organized

**Comment by babaktaremi** [comment]
> The roadmap xMind/Image need to be updated as well. I appreciate if you do it, otherwise I can do that.

Just Updated xMind/Image
Also Categorized the added things to more organized

**Review by babaktaremi — COMMENTED** [review]
Just Updated xMind/Image
Also Categorized the added things to more organized

File: ReadMe.md [hunk 2]
```diff
@@ -117,6 +117,9 @@ If you like or are using this project to learn or start your solution, please gi
        2. [Memcached](https://memcached.org)
     - Entity Framework 2nd Level Cache
        - [EFCoreSecondLevelCacheInterceptor](https://github.com/VahidN/EFCoreSecondLevelCacheInterceptor)
+    - Application-Level
```
Comment (MoienTajik) [inline ReadMe.md:120]: Good idea! In this case, I think EFCore 2nd level cache should be moved under this category as well.

**Review by MoienTajik — COMMENTED** [review]
Could you please compress the generated image via https://tinypng.com to reduce its size?

File: ReadMe.md (hunk 2 above)
Comment (babaktaremi) [reply ReadMe.md:118]: Done

**Comment by babaktaremi** [comment]
Resized Related PNG
Updated route map and added EF Core 2nd level cache to Application-Level Category (Both in ReadME.md and xMind/PNG files)

**Review by MoienTajik — APPROVED** [review]

**Comment by MoienTajik** [comment]
Great! Thanks for your contribution. 💙

**Pull request merged**